packages/maker/src/api/defineObject.ts

```diff
@@ -122,7 +123,15 @@ function convertType(
       return {
         type: "timestamp",
       };
-
+    case typeof t.type === "object": {
+      return {
+        type: "struct",
+        structFieldTypes: Object.entries(t.type).map(([apiName, dataType]) => ({
+          apiName,
+          dataType: { type: dataType } as ObjectPropertyType,
```

Comment:
This cast should also be fine because the maker package only uses this for primary key types, which are limited to a select few primitive types